File: WebSocketServer/Modules/AuthModule.cs
using System;
using GameSystem.DAO;
using GameSystem.DbMock;
using GameSystem.Models;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using WebSocketSharp.Server;

namespace GameSystem.Modules
{
    public class AuthModule
    {
        public void Dispach(Request request, Server webSocket)
        {
            dynamic response = null;
            switch (request.Cmd)
            {
                case "Login":
                    response = Authorize(request);
                    break;
                case "Logout":
                    break;
                case "Registration":
                    response = Registration(request);
                    break;
            }
            if (response != null)
            {
                response.Module = "AuthModule";
                string json = JsonConvert.SerializeObject(response);
                webSocket.SendResponse(json);
            }
        }

        private Response Authorize(Request request)
        {
            Response response = new Response { Module = "AuthModule", Cmd = "Login", Token = "" };
            if (request.UserStatus == UserStatus.Authorized)
            {
                string login = TokenProvider.GetLogin(request.Token);
                response.Token = TokenProvider.GetToken(login);
            }
            else
            {
                IUserAccountDAO db = new UserAccountDAO_Mock();
                string login = request.Args.Login;
                string password = request.Args.Password;
                UserAccount account = db.GetUserByLogin(login);
                if (account == null)
                {
                    response.Args = new { Error = "invalid login" };
                }
                else
                {
                    if (account.Password == password)
                    {
                        string token = TokenProvider.GetToken(account.Login);
                        response.Token = token;
                    }
                    else
                    {
                        response.Args = new { Error = "invalid password" };
                    }
                }
            }
            return response;
        }

        private Response Registration(Request request)
        {
            Response response = new Response { Module = "AuthModule", Cmd = "Registration", Token = "" };
            IUserAccountDAO db = new UserAccountDAO_Mock();
            string login = request.Args.Login;
            string password = request.Args.Password;
            string email = request.Args.Email;
            UserAccount account = new UserAccount {Login = login, Password = password, Email = email};
            db.Create(account);
            string token = TokenProvider.GetToken(account.Login);
            response.Token = token;
            return response;
        }
    }
}

File: WebClient/Scripts/Modules/AuthModule.js
class AuthModule {

    Dispach(response) {
        switch (response.Cmd) {
            case "Login":
                console.log("Login. response token = " + response.Token);
                TokenProvider.saveToken(response.Token);
                break;
            case "Logout":
                break;
            case "Registration":
                console.log("Registration. response token = " + response.Token);
                TokenProvider.saveToken(response.Token);
                break;
        }
    }

    static getAuthData() {
        var login = $("#authLogin");
        var psw = $("#authPassword");
        return {
            Login: login.val(),
            Password: psw.val()
        }
    }

    static getRegistrationData() {
        var login = $("#registrationLogin");
        var psw = $("#registrationPassword");
        var email = $('#email');
        return {
            Login: login.val(),
            Password: psw.val(),
            Email: email.val()
        }
    }
}
